DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1CompareVertexSoA.cc

```diff
@@ -0,0 +1,156 @@
+// -*- C++ -*-
+// Package:    SiPixelPhase1CompareVertexSoA
+// Class:      SiPixelPhase1CompareVertexSoA
+//
+/**\class SiPixelPhase1CompareVertexSoA SiPixelPhase1CompareVertexSoA.cc 
+*/
+//
+// Author: Suvankar Roy Chowdhury
+//
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+// DQM Histograming
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "CUDADataFormats/Vertex/interface/ZVertexHeterogeneous.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+
+class SiPixelPhase1CompareVertexSoA : public DQMEDAnalyzer {
+public:
+  using IndToEdm = std::vector<uint16_t>;
+  explicit SiPixelPhase1CompareVertexSoA(const edm::ParameterSet&);
+  ~SiPixelPhase1CompareVertexSoA() override = default;
+  void bookHistograms(DQMStore::IBooker& ibooker, edm::Run const& iRun, edm::EventSetup const& iSetup) override;
+  void analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  const edm::EDGetTokenT<ZVertexHeterogeneous> tokenSoAVertexCPU_;
+  const edm::EDGetTokenT<ZVertexHeterogeneous> tokenSoAVertexGPU_;
+  const edm::EDGetTokenT<reco::BeamSpot> tokenBeamSpot_;
+  const std::string topFolderName_;
+  const float dzCut_;
+  MonitorElement* hnVertex;
+  MonitorElement* hx;
+  MonitorElement* hy;
+  MonitorElement* hz;
+  MonitorElement* hchi2;
+  MonitorElement* hchi2oNdof;
+  MonitorElement* hptv2;
+  MonitorElement* hntrks;
+};
+
+//
+// constructors
+//
+
+SiPixelPhase1CompareVertexSoA::SiPixelPhase1CompareVertexSoA(const edm::ParameterSet& iConfig)
+    : tokenSoAVertexCPU_(consumes<ZVertexHeterogeneous>(iConfig.getParameter<edm::InputTag>("pixelVertexSrcCPU"))),
+      tokenSoAVertexGPU_(consumes<ZVertexHeterogeneous>(iConfig.getParameter<edm::InputTag>("pixelVertexSrcGPU"))),
+      tokenBeamSpot_(consumes<reco::BeamSpot>(iConfig.getParameter<edm::InputTag>("beamSpotSrc"))),
+      topFolderName_(iConfig.getParameter<std::string>("TopFolderName")),
```

Comment:
unless this is a convention of sorts among DQM modules, could you change `"TopFolderName"` to lowercase `"topFolderName"`, for consistency with the other parameters ?

Reply:
done at [76b4e67](https://github.com/cms-sw/cmssw/pull/37619/commits/76b4e679e9fa98f05f5fc7095d00eb4b03926ef5)